kubernetes-informer/src/main/java/io/micronaut/kubernetes/client/informer/resolvers/DefaultInformerApiGroupResolver.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.informer.resolvers;
+
+import io.kubernetes.client.Discovery;
+import io.kubernetes.client.common.KubernetesObject;
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.kubernetes.client.DiscoveryCache;
+import io.micronaut.kubernetes.client.informer.Informer;
+import io.micronaut.kubernetes.client.informer.InformerAnnotationUtils;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+/**
+ * The default implementation of {@link InformerApiGroupResolver}.
+ *
+ * @author Pavol Gressa
+ * @since 3.3
+ */
+@Singleton
+public class DefaultInformerApiGroupResolver implements InformerApiGroupResolver {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultInformerApiGroupResolver.class);
+
+    private final DiscoveryCache discoveryCache;
+
+    public DefaultInformerApiGroupResolver(@Nullable DiscoveryCache discoveryCache) {
+        this.discoveryCache = discoveryCache;
+    }
+
+    @Override
+    @NonNull
+    public String resolveInformerApiGroup(@NonNull AnnotationValue<Informer> informer) {
+        String apiGroup = null;
+
+        Optional<String> apiGroupOptional = informer.get("apiGroup", String.class);
+        if (apiGroupOptional.isPresent()) {
+            apiGroup = apiGroupOptional.get();
+            if (LOG.isTraceEnabled()) {
```

Comment:
Logic can be simplified:

```
         String apiGroup = informer.stringValue("apiGroup").orElse(null);
```

Reply:
well the think is that the [Informer](https://github.com/micronaut-projects/micronaut-kubernetes/blob/informer-refactor/kubernetes-informer/src/main/java/io/micronaut/kubernetes/client/informer/Informer.java)'s attributes has defaults configured, which are ignored when accessed by `*Value` methods.. that's why the `get` is used:

From AnnotationValue:
```
    @Override
    public <T> Optional<T> get(CharSequence member, ArgumentConversionContext<T> conversionContext) {
        Optional<T> result = convertibleValues.get(member, conversionContext);
        if (!result.isPresent()) {
            Object dv = defaultValues.get(member.toString());
            if (dv != null) {
                return ConversionService.SHARED.convert(dv, conversionContext);
            }
        }
        return result;
    }
```

vs

```
    @Override
    public Optional<String> stringValue(@NonNull String member) {
        if (StringUtils.isNotEmpty(member)) {
            Object o = getRawSingleValue(member, valueMapper);
            if (o != null) {
                return Optional.of(o.toString());
            }
        }
        return Optional.empty();
    }
```

If I would use stringValue, then I would have to resolve defaults on two places which basically duplicates the code.